src/octopeer-github/main/DatabaseAdapter.ts

```diff
@@ -0,0 +1,158 @@
+                    console.log(`Initialized DatabaseAdapter(${_url}, session=${self._session})`, self);
+                }
+            })
+            .fail(function(jqXHR, status) {
+                console.log(`[WARN] DatabaseAdapter could not connect to ${self.url}.`, jqXHR);
+            });
+        if (self._debug) {
+            console.log(`Constructed DatabaseAdapter(${_url})`, self);
+        }
+    }
+
+    get url(): string {
+        return this._url;
+    }
+
+    get isInitialized(): boolean {
+        return this._session !== -1;
+    }
+
+    get session(): number {
+        return this._session;
+    }
+
+    /**
+     * Sets the verbose logging.
+     * @param d = true      Whether verbose logging should be used.
+     */
+    public setDebug(d = true) {
+        this._debug = d;
+    }
+
+    // TODO The database does not support elementType yet.
+    /**
+     * Post an event to the database.
+     * @param elementType   Should equal an existing ElementType ID in the database.
+     * @param eventType     Should equal an existing EventType ID in the database.
+     * @param start         Date when the event started.
+     * @param duration      How long the event lasted, in ms.
+     * @param success       Callback, which is called once the call has succeeded.
+     * @param failure       Callback, which is called once the call has failed.
+     */
+    public log(elementType: number, eventType: number, start: Date, duration: number,
+               success: JQueryPromiseCallback<any>, failure: JQueryPromiseCallback<any>): void {
+        const self = this;
+        if (!this.isInitialized) {
+            console.log("[WARN] The database has not been initialized yet!");
+            return;
+        }
+        const ajax = $.ajax(`${this.url}api/events/`, self.createPostData(eventType, start, duration))
+            .done(function(data, status, jqXHR) {
+                if (self._debug) {
+                    console.log(`Call success, status: ${status}`, jqXHR);
+                }
+                success(data, status, jqXHR);
+            })
+            .fail(function(jqXHR, status) {
+                console.log(`[WARN] Call failed, status: ${status}`, jqXHR);
+                failure(jqXHR, status);
+            });
+        if (self._debug) {
+            console.log("Sent out AJAX request: ", ajax);
+        }
+    }
+
+    public logWLine(elementType: number, eventType: number, start: Date, duration: number,
+                    fileName: string, lineNumber: number,
+                    success: JQueryPromiseCallback<any>, failure: JQueryPromiseCallback<any>): void {
+        this.log(elementType, eventType, start, duration, success, failure);
+    }
+
+    /**
+     * Creates a Settings Object that can be used in an AJAX request when posting a new session.
+     * @returns {JQueryAjaxSettings}    A Settings Object that can be used in an AJAX request.
+     */
+    private createPostSession() {
```

Comment:
WebStorm suggests that the following private methods can be static.
If we do so, TSLint starts complaining and undesired behavior pops up, so let's just ignore that suggestion. :P